### ISSUE TITLE
It looks correct but the AC doesn't start...

### ISSUE THREAD

**Issue by Salan54** [thetonto/node-red-contrib-sensibo#9]
Hi,
Thanks for this great node. I was looking for a way to drive my AC directly from node-red without using IFTTT...
I setup the node with my API key and was able to retrieve my 3 ACs.
I was able to retrieve the parameters from the AC in my office through "Sensibo In node":

```
{
  "_msgid": "1592e83.9786318",
  "payload": "success",
  "topic": "",
  "temperature": 23,
  "humidity": 47.2,
  "secondsAgo": 15,
  "time": "2020-10-07T14:48:53.814326Z"
}
```

I was able to send a formatted message to my Office's AC through "Sensibo send node" and I got a success message :

```
{
  "_msgid": "290b9d41.97e062",
  "payload": {
    "status": "success",
    "result": {
      "status": "Success",
      "reason": "UserRequest",
      "acState": {
        "on": true,
        "targetTemperature": 25,
        "temperatureUnit": "C",
        "mode": "heat",
        "fanLevel": "auto"
      },
      "changedProperties": [],
      "id": "CM7xbnaANF",
      "failureReason": null
    }
  },
  "topic": "",
  "on": true,
  "mode": "heat",
  "fanlevel": "auto",
  "targetTemperature": 25
}
```
My problem is that the AC did not start... Here is the msg object I sent to the "Sensibo send node" (through a "Change node" :
```
{
  "_msgid": "fd189b5a.f4f5d8",
  "payload": 1602082711189,
  "topic": "",
  "on": true,
  "mode": "heat",
  "fanlevel": "auto",
  "targetTemperature": 25
}
```
I surely made an error somewhere. Could you give me a clue ? Thanks
Salan54

**Comment by thetonto** [comment]
Hi Salan
Here is the message structure I have on my setup.

{
"_msgid":"8880f845.55a5c8",
"payload":1602125984262,"topic":"",
"on":"true",
"swing":"rangeFull",
"fanLevel":"quiet",
"targetTemperature":23,
"mode":"heat"
}
which fairly well match's.  Try just the on setting.  Also do you get anything back from the send node?

**Comment by beccsjb** [comment]
I to am having the same problem. 

thetonoto : i have copied your message exactly, I inject it, i can hear the AC beep, but nothing. No AcState changes. (i note that control via sensibo app does work)

Its the beep thats getting me, its obviously trying to send some kind of data, just the wrong data.

Any clues or ideas?

**Comment by thetonto** [comment]
Hi all
I've exported out a test node I just created.  One thing I definitely noticed and not entirely sure why this is happening is that the msg.on must be set as a string (not a binary).  When I was setting is binary true the unit would beep but not process it.  When changed back to a string and set to true it turned on.



`[{"id":"3b74c57f.45f0ca","type":"inject","z":"71d93227.e3061c","name":"","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":160,"y":500,"wires":[["d4634689.bc64c8"]]},{"id":"d4634689.bc64c8","type":"change","z":"71d93227.e3061c","name":"Set Values","rules":[{"t":"set","p":"on","pt":"msg","to":"true","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":350,"y":500,"wires":[["9fb99922.03e5c8"]]},{"id":"9fb99922.03e5c8","type":"sensibo send","z":"71d93227.e3061c","sensiboAPI":"51203865.1cd298","name":"Send to Living Room","pod":"","podSelector":"","x":600,"y":500,"wires":[["d90a3511.a2ed58"]]},{"id":"d90a3511.a2ed58","type":"debug","z":"71d93227.e3061c","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":810,"y":500,"wires":[]},{"id":"51203865.1cd298","type":"sensibo-config","z":"","senAPI":""}]`

**Comment by beccsjb** [comment]
Thanks mate, that did the trick for the turning on atleast, haha.

Still a bit stumped on the other features, have tried setting them via a change node in the same manner and no dice. I'm trying to adapt my previous http request node method to using your node, a few subtle differences.

Could you possibly share a flow; Setting on = true, mode= cool, settemperature=23, fan=auto or something similar, so i can get a better feel for it? Thanks :)

**Comment by thetonto** [comment]
Hi beccsjb.   
Was just thinking to do that.  Good to head the on / off is working.  Thats a big start.

`[{"id":"fd80b3ba.7bb18","type":"inject","z":"71d93227.e3061c","name":"Turn on Aircon","props":[{"p":"payload"},{"p":"topic","vt":"str"}],"repeat":"","crontab":"","once":false,"onceDelay":0.1,"topic":"","payload":"","payloadType":"date","x":180,"y":260,"wires":[["827bed43.54801"]]},{"id":"827bed43.54801","type":"change","z":"71d93227.e3061c","name":"Set Values","rules":[{"t":"set","p":"on","pt":"msg","to":"true","tot":"str"},{"t":"set","p":"swing","pt":"msg","to":"rangeFull","tot":"str"},{"t":"set","p":"fanLevel","pt":"msg","to":"quiet","tot":"str"},{"t":"set","p":"targetTemperature","pt":"msg","to":"23","tot":"num"},{"t":"set","p":"mode","pt":"msg","to":"heat","tot":"str"}],"action":"","property":"","from":"","to":"","reg":false,"x":350,"y":260,"wires":[["681f2f3.eea5ad"]]},{"id":"681f2f3.eea5ad","type":"sensibo send","z":"71d93227.e3061c","sensiboAPI":"51203865.1cd298","name":"Send to Living Room","pod":"xx","podSelector":"","x":580,"y":260,"wires":[["3000fd6a.c54042"]]},{"id":"3000fd6a.c54042","type":"debug","z":"71d93227.e3061c","name":"","active":true,"tosidebar":true,"console":false,"tostatus":false,"complete":"true","targetType":"full","statusVal":"","statusType":"auto","x":820,"y":260,"wires":[]},{"id":"51203865.1cd298","type":"sensibo-config","z":"","senAPI":"xx"}]`

Flow I used to turn on my aircon and heat to 23 degree's.

**Comment by beccsjb** [comment]
Legend! That worked, spot on.

I was previously working with objects and payloads, i see now that your node is simply setting the topics with payloads.

Thanks heaps

**Comment by vlogicau** [comment]
I can confirm that if you inject a simple msg.on and set it as binary, off works but on doesn't. If you instead use a string type and send "true" or "false" then both on and off work as expected.

Thanks for a great plugin.

**Comment by thetonto** [comment]
Have found that there is an issue with the underlying library of requests which this node uses.  I am slowing working on an update to replace the deprecated library and will try and fix this issue at the same time.

**Comment by thetonto** [comment]
Fixed now with version 0.5.0.  Binary can now be used for setting the state of On.
Error was that the code was only checking for a string.  Now checks for both.
